Library/Homebrew/test/cleaner_spec.rb

```diff
@@ -136,6 +136,26 @@
 
       expect(file).not_to exist
     end
+
+    it "removes 'info/**/dir' files except for 'info/<name>/dir'" do
+      file = f.info/"dir"
+      arch_file = f.info/"i686-elf/dir"
+      name_file = f.info/"#{f.name}/dir"
+
+      f.info.mkpath
+      (f.info/"i686-elf").mkpath
+      (f.info/ + f.name.to_s).mkpath
```

Comment:
Or, maybe more simply:
```suggestion
      arch_file.dirname.mkpath
      name_file.dirname.mkpath
```